geode-core/src/main/java/org/apache/geode/internal/cache/DiskStoreImpl.java

```diff
@@ -4387,6 +4411,10 @@ public void waitForDelayedWrites() {
 
   private Future<?> executeTask(final Runnable runnable, ExecutorService executor) {
     // schedule another thread to do it
+    if (executor.isShutdown()) {
+      logger.warn("Submitting task to shutdown pool");
+      return null;
```

Comment:
In general, we need to try to avoid returning nulls (especially in newer code). There may be a proper way to handle submission of execution after shutdown other than returning `null` for the Future such as throwing RejectedExecutionException. I would dig through ExecutorService interface and maybe a JDK impl such as ThreadPoolExecutor. Beyond that, I'm not really familiar with DiskStore code so I'll tag a couple more folks for review.

Reply:
I agree with you, but this is aligned with current implementation. Next method in line "executeDiskStoreTask" is also returning null, in case RejectedExecutionException is caught, when submiting task. Or should we update both methods, and remove this null logic.